aws_batch/pipeline.py

```diff
@@ -468,9 +468,24 @@ def return_merged_dict(dict1, dict2):
 
 def execute_command(command):
     print command
+    sys.stdout.flush()
     output = subprocess.check_output(command, shell=True)
+    print output
+    sys.stdout.flush()
     return output
 
+def execute_command_realtime_stdout(command, progress_file=''):
+    print command
+    sys.stdout.flush()
+    process2 = subprocess.Popen("touch %s ; tail -f %s" % (progress_file, progress_file), shell=True)
+    process = subprocess.Popen(command, shell=True, stdout=subprocess.PIPE)
+    while True:
+        line = process.stdout.readline().rstrip()
+        if not line: break
+        print line
+    process.wait()
+    process2.kill()
```

Comment:
Pretty cool, tho as we discussed the lines can get broken and interleaved in this technique.   Also, for output to be emitted in realtime you'd need the unbuffering trick.  As is, your output will not flush after each line.

Of course, if you do unbuffer, then you could just
```
tail = subprocess.Popen("touch %s ; tail -f %s" % (progress_file, progress_file), shell=True)
try:
  subprocess.check_call(command, shell=True)
finally:
  subprocess.kill(tail)
```

Reply:
Thank you @boris-dimitrov ! Changed it in charles/verbose-logs-2